### PR TITLE
PR: Fix getting help from the console for dot objects and remove redundant info from completion hints

### LINKED ISSUE #11821: Completion doc widget and editor/console to help improvements
- [x]  `Ctrl + I` not working for elements with `.` (for example `np.sin`)
- [x] ~~Doc tooltip from current completion not being updated (for example with `pd.Categorical`)~~ I was unable to reproduce this
- [x] Remove `ufunc` signature from completion doc tooltip (for example with `numpy.sign`, `numpy.sin`, `numpy.signbit`, etc).

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #11821 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/widgets/mixins.py [hunk 1]
```diff
@@ -1359,9 +1363,17 @@ def set_help_enabled(self, state):
         self.help_enabled = state
 
     def inspect_current_object(self):
-        text = self.get_current_word(help_req=True)
```
Comment (ccordoba12) [inline spyder/widgets/mixins.py]: Is `req=True` still necessary in `get_current_word`? If not, please remove it.

File: spyder/widgets/mixins.py [hunk 2]
```diff
@@ -1359,9 +1363,17 @@ def set_help_enabled(self, state):
         self.help_enabled = state
 
     def inspect_current_object(self):
-        text = self.get_current_word(help_req=True)
-        if text:
-            self.show_object_info(text, force=True)
+        regex_word = ''
+        start_text = self.get_text('sol', 'cursor')
+        tl1 = re.findall(r'([a-zA-Z_]+[^\s^\(]*)', start_text)
```
Comment (ccordoba12) [inline spyder/widgets/mixins.py]: Does this capture unicode variables? Please remember that unicode identifiers is valid in Python 3.

File: spyder/widgets/mixins.py [hunk 3]
```diff
@@ -1359,9 +1363,17 @@ def set_help_enabled(self, state):
...
-        text = self.get_current_word(help_req=True)
-        if text:
-            self.show_object_info(text, force=True)
+        regex_word = ''
+        start_text = self.get_text('sol', 'cursor')
+        tl1 = re.findall(r'([a-zA-Z_]+[^\s^\(]*)', start_text)
+        if tl1 and start_text.endswith(tl1[-1]):
+            regex_word += tl1[-1]
+        end_text = self.get_text('cursor', 'eol')
+        tl2 = re.findall(r'([^\s^\(]+[0-9a-zA-Z_\.]*)', end_text)
```
Comment (ccordoba12) [inline spyder/widgets/mixins.py]: How many words with dots this regexp can capture? Please remember that we can have things like `np.linalg.norm` too.

Also, perhaps this should only apply to Python and not to other languages (e.g. the equivalent of `.` in R is `$`).

Comment (ccordoba12) [reply spyder/widgets/mixins.py]: Also, why can't we use here the current text from the cursor to the left, until an space is found? That seems simpler to me than using regexp's.

File: spyder/widgets/mixins.py (hunk 1 above)
Comment (dalthviz) [reply spyder/widgets/mixins.py]: Yep the `help_req` is needed for the help request made from the `Editor`

File: spyder/widgets/mixins.py (hunk 3 above)
Comment (dalthviz) [reply spyder/widgets/mixins.py]: I tried that approach at first, but then I realized that I needed also to get sometimes characters to the right of the cursor, and I wasn't sure of the logic to follow to get does characters and then mix both (left and right selected texts).

I will recheck that approach and see what is better

File: spyder/widgets/mixins.py (hunk 2 above)
Comment (dalthviz) [reply spyder/widgets/mixins.py]: Will check

File: spyder/widgets/mixins.py (hunk 3 above)
Comment (ccordoba12) [reply spyder/widgets/mixins.py]: Thanks @dalthviz!

File: spyder/app/tests/test_mainwindow.py [hunk 4]
```diff
@@ -431,6 +431,45 @@ def test_get_help_combo(main_window, qtbot):
...
+    # Write function name
+    qtbot.keyClicks(control, u'np.sin')
+
+    # Get help
+    control.inspect_current_object()
+
+    # Check that a expected text is part of the page
+    qtbot.waitUntil(
+        lambda: check_text(webpage, "Trigonometric sine, element-wise."),
+        timeout=6000)
```
Comment (ccordoba12) [inline spyder/app/tests/test_mainwindow.py]: Also, below this, please add a test for the aaʹbb function present in `script_unicode.py`, to test if we can show help for objects that contain unicode chars.

File: spyder/widgets/mixins.py [hunk 5]
```diff
@@ -971,14 +1017,24 @@ def is_space(move):
                 return not to_text_string(curs.selectedText()).strip()
 
             def is_special_character(move):
```
Comment (ccordoba12) [inline spyder/widgets/mixins.py]: Please add a docstring to this function. What characters is it supposed to catch?

File: spyder/widgets/mixins.py [hunk 6]
```diff
@@ -971,14 +1017,24 @@ def is_space(move):
...
+                text_cursor = to_text_string(curs.selectedText()).strip()
+                return text_cursor == '.'
+
             if help_req:
-                if is_special_character(QTextCursor.NoMove):
+                if (is_special_character(QTextCursor.NoMove) and
+                        not is_dot(QTextCursor.PreviousCharacter)):
                     cursor.movePosition(QTextCursor.WordLeft)
+                elif is_dot(QTextCursor.PreviousCharacter):
+                    cursor.movePosition(QTextCursor.NextCharacter)
```
Comment (ccordoba12) [inline spyder/widgets/mixins.py]: Checking for `dot` here makes things specialized for Python. What's the plan for other languages?

File: spyder/app/tests/test_mainwindow.py (hunk 4 above)
Comment (dalthviz) [reply spyder/app/tests/test_mainwindow.py]: See `test_mainwindow.test_get_help_ipython_console_special_characters`

Comment (ccordoba12) [reply spyder/app/tests/test_mainwindow.py]: Ok, I haven't noticed that one :+1:

**Pull request merged**